File: templates/java/src/main/java/fi/solita/botsofbf/javabot/Bot.java
package fi.solita.botsofbf.javabot;

import fi.solita.botsofbf.javabot.dto.*;

import java.util.Optional;

public class Bot {
    public final String botName;
    public final String id;
    public final Map map;

    public Bot(String botName, String id, Map map) {
        this.botName = botName;
        this.id = id;
        this.map = map;
    }

    public Optional<Player> findMe(GameState gameState) {
        return gameState.players().stream().filter(p -> p.name().equals(botName)).findFirst();
    }

    public Move resolveNextMove(GameState gameState) {
        // todo, calculate next move
        Position myCurrentPos = findMe(gameState).map(Player::position).orElseThrow();

        if (Math.random() > 0.5) {
            return Move.LEFT;
        }
        else {
            return Move.RIGHT;
        }
    }
}

File: templates/python/bot.py
import random


class Bot:
    def __init__(self, name: str, id_: str):
        self.name = name
        self.id_ = id_

    def find_me(self, game_state):
        for player in game_state["players"]:
            if player["name"] == self.name:
                return player
        return None

    def resolve_next_move(self, game_state):
        player = self.find_me(game_state)
        if player is None:
            raise Exception("Player not found")
        current_pos = player["position"]

        if random.random() > 0.5:
            return "LEFT"
        else:
            return "RIGHT"
